File: 20240523-GrenlandWeb/DotNET/SUT/UserDetailsService.cs
namespace SUT;

public class UserDetailsService(IDatabase database) : IUserDetailsService
{
    readonly IDatabase _database = database;

    public Task Register(string firstName, string lastName, string socialSecurityNumber, Guid userId)
        => _database.GetCollectionFor<UserDetails>().InsertOneAsync(new(Guid.NewGuid(), userId, firstName, lastName, socialSecurityNumber));
}

File: 20240523-GrenlandWeb/TypeScript/src/UserDetailsService.ts
import { Guid } from './Guid';
import { IDatabase } from './IDatabase';
import { IUserDetailsService } from './IUserDetailsService';
import { UserDetails } from './UserDetails';

export class UserDetailsService extends IUserDetailsService {
    constructor(readonly _database: IDatabase) {
        super();
    }

    async register(userId: Guid, firstName: string, lastName: string, socialSecurityNumber: string): Promise<void> {
        await this._database.getCollectionFor(UserDetails).insertOne({ id: userId, firstName, lastName, socialSecurityNumber } as any);
    }
}
